pkg/auth/providers/githubapp/github_client.go

```diff
@@ -0,0 +1,425 @@
+		return nil, err
+	}
+
+	return result.Items, nil
+}
+
+// searchTeams searches for teams that match the search term in the organizations the access token has access to.
+// At the moment it only does a case-insensitive prefix match on the team's name.
+func (g *GClient) searchTeams(searchTerm, githubAccessToken string, config *mgmtv3.GithubAppConfig) ([]Account, error) {
+	orgs, err := g.getOrgs(githubAccessToken, config)
+	if err != nil {
+		return nil, err
+	}
+
+	lowerSearchTerm := strings.ToLower(searchTerm)
+
+	var matches, teams []Account
+	for _, org := range orgs {
+		teams, err = g.getOrgTeams(githubAccessToken, config, org)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, team := range teams {
+			if !strings.HasPrefix(strings.ToLower(team.Name), lowerSearchTerm) {
+				continue
+			}
+
+			matches = append(matches, team)
+		}
+
+	}
+
+	return matches, nil
+}
+
+func (g *GClient) getUserOrgByID(id string, githubAccessToken string, config *mgmtv3.GithubAppConfig) (Account, error) {
+	url := g.getURL("USER_INFO", config) + "/" + id
+
+	b, _, err := g.getFromGithub(githubAccessToken, url)
+	if err != nil {
+		logrus.Errorf("Github getUserOrgById: GET url %v received error from github, err: %v", url, err)
+		return Account{}, err
+	}
+	var githubAcct Account
+
+	if err := json.Unmarshal(b, &githubAcct); err != nil {
+		logrus.Errorf("Github getUserOrgById: error unmarshalling response, err: %v", err)
+		return Account{}, err
+	}
+
+	return githubAcct, nil
+}
+
+// URLEncoded encodes the string
+func URLEncoded(str string) string {
+	u, err := url.Parse(str)
+	if err != nil {
+		logrus.Errorf("Error encoding the url: %s, error: %v", str, err)
+		return str
+	}
+	return u.String()
+}
+
+func (g *GClient) postToGithub(url string, form url.Values) ([]byte, error) {
+	req, err := http.NewRequest("POST", url, strings.NewReader(form.Encode()))
+	if err != nil {
+		logrus.Error(err)
+	}
+	req.PostForm = form
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("Accept", "application/json")
+	resp, err := g.httpClient.Do(req)
+	if err != nil {
+		logrus.Errorf("Received error from github: %v", err)
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	// Check the status code
+	switch resp.StatusCode {
+	case 200:
+	case 201:
+	default:
+		var body bytes.Buffer
+		io.Copy(&body, resp.Body)
+		return nil, fmt.Errorf("request failed, got status code: %d. Response: %s",
+			resp.StatusCode, body.Bytes())
+	}
+	return io.ReadAll(resp.Body)
+}
+
+func (g *GClient) getFromGithub(githubAccessToken string, url string) ([]byte, string, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, "", err
+	}
+	req.Header.Add("Authorization", "token "+githubAccessToken)
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("user-agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36)")
```

Comment:
Q: Is there any reason to be that specific with the `user-agent`?

Reply:
Not sure why that's done...this is pretty much the same as the `github` provider.